main.js

```diff
@@ -0,0 +1,310 @@
+  $('.spinner').removeClass('hide')
+  //pass location query and true for isDefault because we're defining this as the default location
+  getLocationCoordinates(defaultLocation, true)
+}
+
+//use geolocation data to find locality name
+$('.locate-me').on('click', function() {
+  if ("geolocation" in navigator) {
+    //show a spinner while fetching data from the API
+    $('.weekly-forecast').addClass('hide')
+    $('.daily-forecast').addClass('hide')
+    $('.spinner').removeClass('hide')
+    navigator.geolocation.getCurrentPosition(function(position) {
+      locationLatitude = position.coords.latitude
+      locationLongitude = position.coords.longitude
+      getLocationAddress(locationLatitude, locationLongitude)
+    });
+  } else {
+    alert('location services not available')
+  }
+})
+
+//on search get location information and pass it to the weather APIs
+$('.search').on('click', function() {
+  input = $('#search-query').val();
+  //show a spinner while fetching data from the API
+  $('.weekly-forecast').addClass('hide')
+  $('.daily-forecast').addClass('hide')
+  $('.spinner').removeClass('hide')
+  //pass location query and false because we are not defining this as the default location
+  getLocationCoordinates(input, false)
+})
+
+//set a default location and save it in localStorage, by passing this location to getLocationCoordinates with boolean true we set it as the default location
+$('body').on("click", ".default-location", function() {
+  var input = $('#search-query').val();
+  localStorage.setItem('location', JSON.stringify(input))
+  getLocationCoordinates(input, true)
+})
+
+//used in geolocation: gets a location name from the location address data to pass to the getLocationCoordinates function to get weather data
+var parseLocation = function(data) {
+  //iterates through location address data to find the item that gives the locality or sublocality name to pass to the getLocationCoordinates function. If no name, alerts an error and hides the spinner
+  var locationName
+  for (i = 0; i < data.results.length; i++) {
+    for (j = 0; j < data.results[i].types.length; j++) {
+      if (data.results[i].types[j] == "locality" || data.results[i].types[j] == "sublocality") {
+        locationName = data.results[i].formatted_address;
+        break;
+      }
+    }
+  }
+  if (!locationName) {
+    $('.spinner').addClass('hide')
+    alert("Geolocation error, please try again.")
+  }
+  $('#search-query').val(locationName)
+  getLocationCoordinates(locationName, false)
+}
+//pass search query to a Google maps API to get full name, latitude, and longitude then pass latitude and longitude to the openweather API to get weather data
+var getWeather = function(data, isDefault) {
+  $('.weekly-forecast').removeClass('hide')
+  $('.daily-forecast').removeClass('hide')
+  $('.spinner').addClass('hide')
+  //if we're getting weather for the default location, we set the default location name
+  if (isDefault) {
```

Comment:
Why does this need to be a parameter, whether or not its true, the logic is the same, right?  The data determines if its the default location, not what is clicked on.